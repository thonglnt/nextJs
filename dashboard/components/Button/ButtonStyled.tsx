import { Button, styled } from '@mui/material'

// Components
import { IButtonProps } from '@components/Button'

export const ButtonStyled = styled(Button)<IButtonProps>`
  &:disabled {
    cursor: no-drop;
  }
`
