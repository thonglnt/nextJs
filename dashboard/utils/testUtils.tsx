import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
