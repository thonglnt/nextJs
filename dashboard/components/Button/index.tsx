'use client'

import { memo } from 'react'
import { ButtonProps } from '@mui/material'

// Styles
import { ButtonStyled } from './ButtonStyled'

export interface IButtonProps extends ButtonProps {
  children: React.ReactNode | string
}

const Button = ({ children = 'Button', onClick, ...props }: IButtonProps) => {
  return (
    <ButtonStyled
      onClick={onClick}
      {...props}
      sx={{
        padding: {
          xs: '6px 16px',
          sm: '8px 20px',
          md: '11px 24px',
        },
        fontSize: '14px',
      }}
    >
      {children}
    </ButtonStyled>
  )
}

export default memo(Button)
