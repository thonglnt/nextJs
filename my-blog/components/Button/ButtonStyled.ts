import { Button, styled } from '@mui/material'

// Components
import { IButtonProps } from '@components/Button'

export const ButtonStyled = styled(Button)<IButtonProps>`
  cursor: pointer;
  color: ${({ theme }) => theme.palette.grey[800]};
  font-weight: 700;
  background-color: ${({ theme }) => theme.palette.grey[200]};
  &:hover {
    color: ${({ theme }) => theme.palette.common.white};
    background-color: ${({ theme }) => theme.palette.grey[600]};
  }
`
