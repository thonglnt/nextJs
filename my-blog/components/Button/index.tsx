'use client'

import { memo } from 'react'
import { ButtonProps } from '@mui/material'

// Styles
import { ButtonStyled } from './ButtonStyled'

export interface IButtonProps extends ButtonProps {
  children: React.ReactNode | string
}

const Button = ({ children = 'Button', onClick, ...props }: IButtonProps) => {
  return (
    <ButtonStyled onClick={onClick} {...props}>
      {children}
    </ButtonStyled>
  )
}

export default memo(Button)
