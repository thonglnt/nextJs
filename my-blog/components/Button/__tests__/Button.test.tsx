// Utils
import { fireEvent, render, screen } from '@utils/testUtils'
import Button, { IButtonProps } from '..'

// Components

describe('Button Component', () => {
  const handleClick = jest.fn()

  const mockProps: IButtonProps = {
    children: 'Add To Cart',
    onClick: handleClick,
  }

  it('Should show match Button DOM Snapshot', () => {
    const { container } = render(<Button {...mockProps} />)

    expect(container).toMatchSnapshot()
  })

  it('Should call onClick on button ', () => {
    render(<Button {...mockProps} />)

    const button = screen.getByRole('button')

    fireEvent.click(button)

    expect(handleClick).toBeCalled()
    expect(handleClick).toBeCalledTimes(1)
  })
})
