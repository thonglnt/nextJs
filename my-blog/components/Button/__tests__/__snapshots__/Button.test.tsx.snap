// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should show match Button DOM Snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1qtdmvj-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Add To Cart
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
